### PR TITLE
Add pre-releases and caching to Mac builds

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Adds caching to Mac builds
Adds development pre-releases
Adds additional linters

### CODE REVIEW

File: tools/additional_lint [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    line: int | None = None
+
+    def __str__(self) -> str:
+        prefix = str(self.path.relative_to(REPO_DIR))
+        if self.line is not None:
+            prefix += f":{self.line}"
+        return f"{prefix}: {self.message}"
+
+
+def get_files() -> Iterable[Path]:
```
Comment (lahm86) [inline tools/additional_lint:47]: Is it possible to exclude the following directories here?

`tools/config/TR1X_ConfigTool/obj`
`tools/config/TR1X_ConfigTool/bin`

These will contain auto-generated files that trigger warnings e.g.

```
tools/config/TR1X_ConfigTool/obj/project.assets.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/obj/TR1X_ConfigTool.csproj.nuget.dgspec.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/obj/Release/net6.0-windows/win-x64/App.g.cs:36: trailing whitespace
...
tools/config/TR1X_ConfigTool/bin/Release/net6.0-windows/win-x64/TR1X_ConfigTool.deps.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/bin/Release/net6.0-windows/win-x64/TR1X_ConfigTool.runtimeconfig.json: missing newline character at end of file
```

**Review by lahm86 — COMMENTED** [review]
Looks good, just some annoying VS quirks to work around.

File: tools/additional_lint (hunk 1 above)
Comment (rr-) [reply tools/additional_lint:47]: Done, please LMK if it works as intended now.

Comment (lahm86) [reply tools/additional_lint:47]: Works perfectly 👍

**Review by lahm86 — APPROVED** [review]

**Review by walkawayy — APPROVED** [review]
Seems good on my end.

**Pull request merged**